integration-tests/tests/run-rstudio.js

```diff
@@ -0,0 +1,66 @@
+// This test is owned by the Interactive Analysis (IA) Team.
+const _ = require('lodash/fp')
+const { withRegisteredUser, withBilling, withWorkspace, performAnalysisTabSetup } = require('../utils/integration-helpers')
+const {
+  click, clickable, findElement, noSpinnersAfter, fillIn, findIframe, findText, dismissNotifications, getAnimatedDrawer, image, input
+} = require('../utils/integration-utils')
+const { registerTest } = require('../utils/jest-utils')
+
+
+const rFileName = 'test-rmd'
+
+const testRunRStudioFn = _.flow(
+  withWorkspace,
+  withBilling,
+  withRegisteredUser
+)(async ({ workspaceName, page, testUrl, token }) => {
+  await performAnalysisTabSetup(page, token, testUrl, workspaceName)
+
+  // Create analysis file
+  await click(page, clickable({ textContains: 'Start' }))
+  await findElement(page, getAnimatedDrawer('Select an application'))
+  await click(page, image({ text: 'Create new R markdown file' }))
+  await fillIn(page, input({ placeholder: 'Enter a name' }), rFileName)
+  await noSpinnersAfter(page, { action: () => click(page, clickable({ text: 'Create Analysis' })) })
+
+  // Close the create cloud env modal that pops up
+  await noSpinnersAfter(page, {
+    action: () => findText(page, 'A cloud environment consists of application configuration, cloud compute and persistent disk(s).')
+  })
+
+  await click(page, clickable({ textContains: 'Close' }))
+
+  // Navigate to analysis launcher
+  await findElement(page, clickable({ textContains: rFileName }))
+  await click(page, clickable({ textContains: rFileName }))
+  await dismissNotifications(page)
+
+  await noSpinnersAfter(page, {
+    action: () => click(page, clickable({ textContains: 'Open' }))
+  })
+
+  //Create a cloud env from analysis launcher
+  await noSpinnersAfter(page, { action: () => click(page, clickable({ text: 'Create' })) })
+  await findElement(page, clickable({ textContains: 'RStudio Environment ( Creating )' }), { timeout: 40000 })
```

Comment:
One thing I've started doing on AoU tests is to create constants for these timeouts, e.g.,
```javascript
const RStudioCreationWaitMs = 40000
```
Then, my whole test timeout becomes something like:
```javascript
timeout: RStudioCreationWaitMs + otherWaitMs + [some buffer]
```

Reply:
Hmm... I'll keep this in mind for other tests that use the same/similar timeouts a lot, but there are only 3 custom timeouts and the purposes are fairly different, so I think keeping the numbers in the test helps readability